File: FestivalJava/Networking/src/main/java/dto/DTOUtils.java
package dto;

import Domain.Artist;
import Domain.Show;
import Domain.Transaction;
import Domain.User;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Sebi on 29-Mar-17.
 *
 * Conversions between model objects and dto objects
 */
public class DTOUtils {
    public static User getUserFromDTO(UserDTO userDTO){
        String username = userDTO.getUsername();
        String password = userDTO.getPassword();
        return new User(username, password);
    }

    public static UserDTO getUserDTO(User user){
        String username = user.getUsername();
        String password = user.getPassword();
        return new UserDTO(username, password);
    }

    public static Artist getArtistFromDTO(ArtistDTO artistDTO){
        Integer id = artistDTO.getId();
        String name = artistDTO.getName();
        return new Artist(id, name);
    }

    public static ArtistDTO getArtistDTO(Artist artist){
        Integer id = artist.getIdArtist();
        String name = artist.getName();
        return new ArtistDTO(id, name);
    }

    public static Show getShowFromDTO(ShowDTO showDTO){
        Integer id = showDTO.getId();
        String location = showDTO.getLocation();
        String date = showDTO.getDate();
        Integer ticketsAvailable = showDTO.getTicketsAvailable();
        Integer ticketsSold = showDTO.getTicketsSold();
        Artist artist = getArtistFromDTO(showDTO.getArtist());
        return new Show(id, location, date,
                ticketsAvailable, ticketsSold, artist);
    }

    public static ShowDTO getShowDTO(Show show){
        Integer id = show.getIdShow();
        String location = show.getLocation();
        String date = show.getDate();
        Integer ticketsAvailable = show.getTicketsAvailable();
        Integer ticketsSold = show.getTicketsSold();
        ArtistDTO artist = getArtistDTO(show.getArtist());
        return new ShowDTO(id, location, date,
                ticketsAvailable, ticketsSold, artist);
    }

    public static Transaction getTransactionFromDTO(TransactionDTO transactionDTO){
        Integer id = transactionDTO.getId();
        String clientName = transactionDTO.getClientName();
        Integer numberOfTickets = transactionDTO.getNumberOfTickets();
        Show show = getShowFromDTO(transactionDTO.getShow());
        return new Transaction(id, clientName, numberOfTickets, show);
    }

    public static TransactionDTO getTransactionDTO(Transaction transaction){
        Integer id = transaction.getIdTransaction();
        String clientName = transaction.getClientName();
        Integer numberOfTickets = transaction.getNumberOfTickets();
        ShowDTO show = getShowDTO(transaction.getShow());
        return new TransactionDTO(id, clientName, numberOfTickets, show);
    }

    public static List<Artist> getListArtistFromDTO(List<ArtistDTO> artists){
        List<Artist> artistList = new ArrayList<>();
        for (ArtistDTO artistDTO : artists){
            artistList.add(getArtistFromDTO(artistDTO));
        }
        return artistList;
    }

    public static List<ArtistDTO> getListArtistDTO(List<Artist> artists){
        List<ArtistDTO> artistDTOList = new ArrayList<>();
        for (Artist artist : artists){
            artistDTOList.add(getArtistDTO(artist));
        }
        return artistDTOList;
    }

    public static List<Show> getListShowFromDTO(List<ShowDTO> shows){
        List<Show> showList = new ArrayList<>();
        for (ShowDTO showDTO : shows){
            showList.add(getShowFromDTO(showDTO));
        }
        return showList;
    }

    public static List<ShowDTO> getListShowDTO(List<Show> shows){
        List<ShowDTO> showDTOList = new ArrayList<>();
        for (Show show : shows){
            showDTOList.add(getShowDTO(show));
        }
        return showDTOList;
    }
}

File: FestivalC#/Festival/Networking/dto/DTOUtils.cs
using Festival.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Networking.dto
{
    public class DTOUtils
    {
        public static User getUserFromDTO(UserDTO userDTO)
        {
            string username = userDTO.Username;
            string password = userDTO.Password;
            return new User(username, password);
        }

        public static UserDTO getUserDTO(User user)
        {
            string username = user.Username;
            string password = user.Password;
            return new UserDTO(username, password);
        }

        public static Artist getArtistFromDTO(ArtistDTO artistDTO)
        {
            int id = artistDTO.Id;
            string name = artistDTO.Name;
            return new Artist(id, name);
        }

        public static ArtistDTO getArtistDTO(Artist artist)
        {
            int id = artist.IdArtist;
            string name = artist.Name;
            return new ArtistDTO(id, name);
        }

        public static Show getShowFromDTO(ShowDTO showDTO)
        {
            int id = showDTO.Id;
            string location = showDTO.Location;
            DateTime date = showDTO.Date;
            int ticketsAvailable = showDTO.TicketsAvailable;
            int ticketsSold = showDTO.TicketsSold;
            Artist artist = getArtistFromDTO(showDTO.Artist);
            return new Show(id, location, date,
                    ticketsAvailable, ticketsSold, artist);
        }

        public static ShowDTO getShowDTO(Show show)
        {
            int id = show.IdShow;
            string location = show.Location;
            DateTime date = show.Date;
            int ticketsAvailable = show.TicketsAvailable;
            int ticketsSold = show.TicketsSold;
            ArtistDTO artist = getArtistDTO(show.Artist);
            return new ShowDTO(id, location, date,
                    ticketsAvailable, ticketsSold, artist);
        }

        public static Transaction getTransactionFromDTO(TransactionDTO transactionDTO)
        {
            int id = transactionDTO.Id;
            string clientName = transactionDTO.ClientName;
            int numberOfTickets = transactionDTO.NumberOfTickets;
            Show show = getShowFromDTO(transactionDTO.Show);
            return new Transaction(id, clientName, numberOfTickets, show);
        }

        public static TransactionDTO getTransactionDTO(Transaction transaction)
        {
            int id = transaction.IdTransaction;
            string clientName = transaction.ClientName;
            int numberOfTickets = transaction.NumberOfTickets;
            ShowDTO show = getShowDTO(transaction.Show);
            return new TransactionDTO(id, clientName, numberOfTickets, show);
        }

        public static List<Artist> getListArtistFromDTO(List<ArtistDTO> artists)
        {
            List<Artist> artistList = new List<Artist>();
            foreach (ArtistDTO artistDTO in artists)
            {
                artistList.Add(getArtistFromDTO(artistDTO));
            }
            return artistList;
        }

        public static List<ArtistDTO> getListArtistDTO(List<Artist> artists)
        {
            List<ArtistDTO> artistDTOList = new List<ArtistDTO>();
            foreach (Artist artist in artists)
            {
                artistDTOList.Add(getArtistDTO(artist));
            }
            return artistDTOList;
        }

        public static List<Show> getListShowFromDTO(List<ShowDTO> shows)
        {
            List<Show> showList = new List<Show>();
            foreach (ShowDTO showDTO in shows)
            {
                showList.Add(getShowFromDTO(showDTO));
            }
            return showList;
        }

        public static List<ShowDTO> getListShowDTO(List<Show> shows)
        {
            List<ShowDTO> showDTOList = new List<ShowDTO>();
            foreach (Show show in shows)
            {
                showDTOList.Add(getShowDTO(show));
            }
            return showDTOList;
        }
    }
}
